test/denormalized_has_many_test.rb

```diff
@@ -141,6 +141,55 @@ def test_deprecated_set_inverse_association_on_cache_hit
     assert_equal item.object_id, associated_record.item.object_id
   end
 
+  def test_returned_records_should_be_readonly_on_cache_hit
+    IdentityCache.with_fetch_read_only_records do
+      Item.fetch(@record.id) # warm cache
+
+      record_from_cache_hit = Item.fetch(@record.id)
+      record_from_cache_hit.fetch_associated_records.each do |record|
+        assert record.readonly?, "Record #{record} wasn't readonly"
+      end
```

Comment:
While the clarity of the error message on test failure is nice, I would probably use the more concise:
```ruby
assert record_from_cache_hit.fetch_associated_records.all? &:readonly?
```